src/zope/securitypolicy/grantinfo.py

```diff
@@ -36,42 +36,35 @@
 @implementer(IGrantInfo)
 class AnnotationGrantInfo(object):
 
-    prinper = prinrole = permrole = {}
-
     def __init__(self, context):
         self._context = context
-        annotations = IAnnotations(context, None)
-        if annotations is not None:
+        annotations = IAnnotations(context, {})
 
-            prinper = annotations.get(prinperkey)
-            if prinper is not None:
-                self.prinper = prinper._bycol  # by principals
+        # by principals
+        prinper = annotations.get(prinperkey)
+        self.prinper = prinper._bycol if prinper is not None else {}
 
-            prinrole = annotations.get(prinrolekey)
-            if prinrole is not None:
-                self.prinrole = prinrole._bycol  # by principals
+        # by principals
+        prinrole = annotations.get(prinrolekey)
+        self.prinrole = prinrole._bycol if prinrole is not None else {}
 
-            roleper = annotations.get(rolepermkey)
-            if roleper is not None:
-                self.permrole = roleper._byrow  # by permission
+        # by permission
+        roleper = annotations.get(rolepermkey)
+        self.permrole = roleper._byrow if roleper is not None else {}
 
-    def __nonzero__(self):
+    def __bool__(self):
         return bool(self.prinper or self.prinrole or self.permrole)
 
+    __nonzero__ = __bool__
```

Comment:
TBH I think this is a Python 3 bugfix that deserves its own line in the changelog.

Reply:
There is one: 
> - Make ``SecurityMap`` and ``AnnotationGrantInfo`` have proper truth
   behaviour on Python 3; previously they were always true.

Reply:
Whoops, sorry about that!